test/test_intensify/test_successive_halving.py

```diff
@@ -0,0 +1,313 @@
+        self.assertEqual(intensifier.min_budget, 1)
+        self.assertEqual(intensifier.max_budget, 10)
+        self.assertEqual(intensifier.init_chal, 8)  # 4 iterations
+        self.assertEqual(intensifier.cutoff_as_budget, True)
+
+    def test_init_3(self):
+        """
+            test parameter initialiations for successive halving - runtime cutoff as budget (no param provided)
+        """
+
+        with self.assertRaises(ValueError):  # runtime cutoff as budget requires min budget to be provided
+            SuccessiveHalving(
+                tae_runner=None, stats=self.stats,
+                traj_logger=TrajLogger(output_dir=None, stats=self.stats),
+                rng=np.random.RandomState(12345), deterministic=True, run_obj_time=False,
+                cutoff=10, instances=[1])
+
+    def test_top_k_1(self):
+        """
+            test _top_k() for configs with same instance-seed-budget keys
+        """
+        intensifier = SuccessiveHalving(
+            tae_runner=None, stats=self.stats,
+            traj_logger=TrajLogger(output_dir=None, stats=self.stats),
+            rng=np.random.RandomState(12345),
+            instances=[1], min_budget=1)
+        self.rh.add(config=self.config1, cost=1, time=1,
+                    status=StatusType.SUCCESS, instance_id=1,
+                    seed=None,
+                    additional_info=None)
+        self.rh.add(config=self.config1, cost=1, time=1,
+                    status=StatusType.SUCCESS, instance_id=2,
+                    seed=None,
+                    additional_info=None)
+        self.rh.add(config=self.config2, cost=2, time=2,
+                    status=StatusType.SUCCESS, instance_id=1,
+                    seed=None,
+                    additional_info=None)
+        self.rh.add(config=self.config2, cost=2, time=2,
+                    status=StatusType.SUCCESS, instance_id=2,
+                    seed=None,
+                    additional_info=None)
+        conf = intensifier._top_k(configs=[self.config2, self.config1],
+                                  k=1, run_history=self.rh)
+
+        self.assertEqual(conf, [self.config1])
+
+    def test_top_k_2(self):
+        """
+            test _top_k() for configs with different instance-seed-budget keys
+        """
+        intensifier = SuccessiveHalving(
+            tae_runner=None, stats=self.stats,
+            traj_logger=TrajLogger(output_dir=None, stats=self.stats),
+            rng=np.random.RandomState(12345),
+            instances=[1, 2], min_budget=1)
+        self.rh.add(config=self.config1, cost=1, time=1,
+                    status=StatusType.SUCCESS, instance_id=1,
+                    seed=None,
+                    additional_info=None)
+        self.rh.add(config=self.config2, cost=10, time=10,
+                    status=StatusType.SUCCESS, instance_id=2,
+                    seed=None,
+                    additional_info=None)
+
+        with self.assertRaises(AssertionError):
```

Comment:
Could you please use `self.assertRaisesRegex` so that it is clear which error message is raised?